compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -898,8 +944,9 @@ public BLangNode transform(UnionTypeDescriptorNode unionTypeDescriptorNode) {
         BLangFiniteTypeNode bLangFiniteTypeNode = (BLangFiniteTypeNode) TreeBuilder.createFiniteTypeNode();
         for (TypeDescriptorNode finiteTypeEl : finiteTypeElements) {
             SingletonTypeDescriptorNode singletonTypeNode = (SingletonTypeDescriptorNode) finiteTypeEl;
-            BLangLiteral literal = createSimpleLiteral(singletonTypeNode.simpleContExprNode(), true);
-            bLangFiniteTypeNode.addValue(literal);
+            BLangExpression literalOrExpression = createLiteralOrExpression(singletonTypeNode.simpleContExprNode(),
```

Comment:
`literalOrExpression` isn't required?

Reply:
It is added to the created `bLangFiniteTypeNode` in the next line right... Didn't quite get what is meant by this, can you elaborate on this.

Reply:
Meant we can just do

```java
bLangFiniteTypeNode.addValue(createLiteralOrExpression(singletonTypeNode.simpleContExprNode(), true));
```